tests/baselines/reference/arrayCast.errors.txt

```diff
@@ -9,8 +8,7 @@ arrayCast.ts(3,23): error TS2352: Conversion of type '{ foo: string; }[]' to typ
     <{ id: number; }[]>[{ foo: "s" }];
                           ~~~
 !!! error TS2352: Conversion of type '{ foo: string; }[]' to type '{ id: number; }[]' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.
-!!! error TS2352:   Type '{ foo: string; }' is not comparable to type '{ id: number; }'.
-!!! error TS2352:     Object literal may only specify known properties, and 'foo' does not exist in type '{ id: number; }'.
+!!! error TS2352:   Object literal may only specify known properties, and 'foo' does not exist in type '{ id: number; }'.

```

Comment:
This error should never have been issued in the first place. It seems like a bug in our element-wise elaboration logic. Notice how [in the following example](https://www.typescriptlang.org/play?#code/FDA2FMBcAIHsGsCGBPaBeaBvaAzWsAuaAImOgF9pEBnLaASwBMiBGAJgGYKBuYAej7Qhw4QD9xEyVOlT+ggMKwAdgDdwAJ2r1lcHNEjIADuGgBybHkLRqkdfSUBzbhVP7Y+oyfMNm0dh2dyVwBbFGgAIxNEaGD6G0R4E0iAY0QAV2oTJXB6SAALDQ9jazScHHpk+nAlSFBUWDV1UERDWgB3XLz9Arh8jQA6aABJPXy46DaaBhrqyG0lRFAAGmhk5UaYPuhwAA9DdXBqLR1Id1M0pXglWDalV3LNSH65YQAFdVhjdQMzJldx2JHewOaZFLwWfBEGx2RyBVzhNIwA4ARzS9AOjFBBmK3iYrE4cOeYCgE0QiBgGAA2hCrKQKABdKi0bB4vycCiU+m8AQiXlicQvRSqDTHJS6MFmGlQ2zAwKc1ynCW43z+OX0kJhSJUGJxSAJJLgVIZLI5PrqCXUUrlSqzOpwRrNVoTTrdEywM2DEbdcaTWj2SCzeaLFZrYXfV3bPYHIEnM4XK43O64dE2Z48oQAFU8ktwkOsMthLgYtGuMDWwUMiHUiHCEDcSpZKoJLjTghEAHlwgArQ0wUC5DSLGJhZR26jGSo4VAJ27QfafDRzQ4rRBKTGmSyuRiwQ7QUuR3VY7PK-EBFvAIA) one of these is has a *missing* property error and the other is an *excess* property error:

```ts
let okay = { foo: "" } as { id: 123 };
//         ~~~~~~~~~~~~~~~~~~~~~~~~~~
// Conversion of type '{ foo: string; }' to type '{ id: 123; }' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.
//   Property 'id' is missing in type '{ foo: string; }' but required in type '{ id: 123; }'.

let waat = [{ foo: "" }] as { id: 123 }[];
//            ~~~
// Conversion of type '{ foo: string; }[]' to type '{ id: 123; }[]' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.
//   Type '{ foo: string; }' is not comparable to type '{ id: 123; }'.
//     Object literal may only specify known properties, and 'foo' does not exist in type '{ id: 123; }'.
```

Reply:
Opened up https://github.com/microsoft/TypeScript/issues/55167

Reply:
Yeah, I wouldn’t ever expect an EPC error on a *type assertion*.